src/test/scala/uk/gov/hmrc/domain/NinoSpec.scala

```diff
@@ -16,61 +16,217 @@
+      "formatted" in {
+        SuffixedNino("CS100700A").formatted shouldBe "CS 10 07 00 A"
+      }
+      "shortened without spaces" in {
+        SuffixedNino("CS100700A").shorten shouldBe ShortNino("CS100700")
+      }
+      "shortened with spaces" in {
+        SuffixedNino("CS 10 07 00 A").shorten shouldBe ShortNino("CS 10 07 00")
+      }
     }
-    "fail with less than 6 middle digits" in { validateNino("AB12345C") should equal (false) }
-    "fail with more than 6 middle digits" in { validateNino("AB1234567C") should equal (false) }
 
-    "fail if we start with invalid characters" in {
-      val invalidStartLetterCombinations = List('D', 'F', 'I', 'Q', 'U', 'V').combinations(2).map(_.mkString("")).toList
-      val invalidPrefixes = List("BG", "GB", "NK", "KN", "TN", "NT", "ZZ")
-      for (v <- invalidStartLetterCombinations ::: invalidPrefixes) {
-        validateNino(v + "123456C") should equal (false)
+    "Creating as a ShortOrSuffixedNino should" - {
+      for ((description, value) <- validNinos) {
+        s"extract to SuffixedNino a valid number $description: '$value'" in {
+          ShortOrSuffixedNino(value) should (be (a [ShortOrSuffixedNino]) and be (a [SuffixedNino]))
+        }
+      }
+      for ((description, value) <- invalidNinos.filter { case(_,v) => v != validNino }) {
+        s"fail with $description: '$value'" in {
+          an[IllegalArgumentException] should be thrownBy ShortOrSuffixedNino(value)
+        }
       }
     }
 
-    "fail if the second letter O" in {
-      validateNino("AO123456C") should equal (false)
+    "Parsing as a ShortOrSuffixedNino should" - {
+      for ((description, value) <- validNinos) {
+        s"extract to ShortNino a valid number $description: '$value'" in {
+          ShortOrSuffixedNino.parse(value).get should (be (a [ShortOrSuffixedNino]) and be (a [SuffixedNino]))
+        }
+      }
+      for ((description, value) <- invalidNinos.filter { case(_,v) => v != validNino }) {
+        s"fail with $description: '$value'" in {
+          ShortOrSuffixedNino.parse(value) should be (empty)
+        }
+      }
     }
   }
 
-  "Creating a Nino" should {
-    "fail if the nino is not valid" in {
-      an [IllegalArgumentException] should be thrownBy Nino("INVALID_NINO")
+  "For ShortNino" - {
+    val validNinos = List(
+      "without spaces" -> validNino,
+      "with spaces" -> "AB 12 34 56"
+    )
+    val invalidNinos =
+      List(
+        "valid number with a suffix" -> (validNino + validSuffix),
+        "valid number with leading space" -> s" $validNino",
+        "valid number with trailing space" -> s"$validNino ",
+        "only one starting letter and 7 total characters" -> "A123456",
+        "only one starting letter and 8 total characters" -> "A1234567",
+        "three starting letters and 8 total characters" -> "ABC12345",
+        "three starting letters and 9 total characters" -> "ABC123456",
+        "lowercase letters" -> "ab123456",
```

Comment:
Can we not accept lowercase letters? Would it not be better to take any input, uppercase it and store the upper cased version of the NINO they entered? I'm just thinking for when we use this ShortNino in a form, a user may enter their NINO in lowercase, in this case the frontend app would have to upper case it before passing it into the ShortNino object.